openbrokerapi/api.py

```diff
@@ -84,6 +86,25 @@ def decorated(*args, **kwargs):
 
         return decorated
 
+    def get_service_by_id(service_id: str):
+        for service in services:
+            if service.id == service_id:
+                return service
+        raise KeyError('service {} not found'.format(service_id))
+
+    def add_service_id_to_async_response(response, service_id: str):
+        patch = False
+        if isinstance(response, ProvisionedServiceSpec):
```

Comment:
We could also add a `@property` `is_async` to ProvisionedServiceSpec, so we avoid here checking for 'isinstance'.

Would also shorter the code.
`patch = response.is_async`

Reply:
It's currently a bit dirty as we call this on separate types of classes, but at least the check (isinstance) is dirty too. I was still looking for a better way to only send Response or Spec classes into this method.